### DISCUSSION TITLE
Does spring cloud supports all the resilience patterns or just circuit breaker alone

### DISCUSSION THREAD

**Discussion by nagkumar** [resilience4j/resilience4j#2005] [General]
https://spring.io/projects/spring-cloud-circuitbreaker from this it says only circuit breaker pattern is supported.. however 

resilience4j supports many others..(such as Circuit Breaker, Rate Limiter, Retry or Bulkhead.)

That way resilience4j is much deeper than the spring circuitbreaker, however, I do read that resilience4j  should be for lighter works.

For resilience needs do we need to use both spring-cloud-circuitbreaker and resilience4j, if only one, which one is the most advanced one to use?

**Comment by RobWin** [comment]
What exactly is your question?

**Reply by nagkumar** [reply]
what factors decide if we need to use spring-cloud-circuit breaker or resilence4j?

**Reply by RobWin** [reply]
If you use Resilience4j directly, you have a direct dependency to packages/classes of resilience4j. If you want to replace it by something else in the future, you have to migrate it by yourself.
If you use Spring Cloud CircuitBreaker, you are using some abstraction of Spring Boot above Resilience4j. It might be simpler to replace Resilience4j at a later point in time.

**Reply by nagkumar** [reply]
>, you are using some abstraction of Spring Boot above Resilience4j.

But https://spring.io/projects/spring-cloud-circuitbreaker Spring Cloud Circuit breaker provides an abstraction across different circuit breaker implementations only...

How about the abstractions for _Rate Limiter, Retry or Bulkhead_ and other resilience-building patterns?

Bit confused it says Cloud Circuit Breaker, but when I see implementations it says Spring Retry.. I would have expected it to be CircuitBreaker Impl not the retry

<img width="790" alt="image" src="https://github.com/resilience4j/resilience4j/assets/332234/847bab6b-b4eb-422c-9174-50d38de23e65">

while in resilence4j I see them as different (may be unrelated) modules that enable resilience...

<img width="106" alt="image" src="https://github.com/resilience4j/resilience4j/assets/332234/7b9b59fa-2226-45b2-bb16-121965e87d38">

**Reply by RobWin** [reply]
Spring Cloud CircuitBreaker is a different project. You need to ask your questions there: https://github.com/spring-cloud/spring-cloud-circuitbreaker
